### ISSUE TITLE
Auto compression for HTTP and use zstd

### ISSUE THREAD

**Issue by uasan** [uNetworking/uWebSockets.js#1039]
Hey.

I think this topic has already been raised, but once again I want to describe how important it is for real performance.

Most HTTP responses need to be compressed, now this has to be done on the Node side, this is not effective, and Chrome already supports the new compression method zstd: https://chromestatus.com/feature/6186023867908096
Safari and Firefox also plan to implement it.

It is ideal for fast (faster than gzip) stream compression, but Node does not have normal zstd streaming libraries.
If uWS implements compression on its side and will check the client Accept-Encoding header, if its value contains `zstd` then compress zstd, if not then `gzip`.

This is something that will really speed up all existing servers that use uWS, thank you.

**Comment by uasan** [comment]
Yes, of course, I had that an explicit method was needed, for example something like this `response.isCompresed(true | false)` by default there should be no compression.

This method will make it possible not to use compression in Node.js, this will be very useful, over time `zstd` can be added later

**Comment by e3dio** [comment]
I use a middleware like this:

```js
app.get('/example.js', serveFile('src/example.js'))
```

`serveFile` sets up a file watcher, runs `brotliCompress` from `node:zlib` on any file changes, stores the raw bytes and compressed bytes in ram, serves correct file according to request headers and sets Etag header. Might publish that code along with some other helpers

**Comment by uasan** [comment]
I will describe my particular case, I need to export data from a database to CSV format via HTTP, the response is 70 MB of text, gzip compresses it to 10 MB, but it takes 2 minutes with 100% processor load, this is a performance disaster, so we really need the ability to compress in uWS

**Comment by uNetworkingAB** [comment]
What I do myself is to just pre-compress everything in RAM when the root folder changes, and serve from pre-compressed RAM. That's really the way to go for HTTP, since it benefits from being state-less unlike WebSockets.

**Comment by uNetworkingAB** [comment]
You already have functions in Node.js to easily compress, and you just add a header. So really nothing to add in the library.

**Comment by uasan** [comment]
@uNetworkingAB you are right, the zstd compression method appeared in the node, but there are 2 options, synchronous, which blocks the process but is not slow, or streaming, which does not block, but is slow, so I suggested adding this to your library.

This only makes sense for dynamic content, for immutable statics, it is more profitable to compress in advance and give out compressed, I absolutely agree with this.
